apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -2137,31 +2140,60 @@ def create_ingestion_dag(
             input_vectors_work_items_per_worker_during_sampling
         )
 
-        # We can't set as default in the function due to the use of `str(threads)`
-        # For consistency we then apply all defaults for resources here.
-        if ingest_resources is None:
-            ingest_resources = {"cpu": str(threads), "memory": "16Gi"}
 
-        if consolidate_partition_resources is None:
-            consolidate_partition_resources = {"cpu": str(threads), "memory": "16Gi"}
+        if(size >= input_vectors_per_work_item):
```

Comment:
Why do we need to have 2 cases here? I would suggest to merge the logic by using min, max computation

```
mem_based_on_size=...
ram = min(16, max(2, mem_based_on_size))
```

Reply:
Refactored, good call!